LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,342 @@
+            }
+        }) {
+            if isEditing {
+                return AnyView(Text(NSLocalizedString("Cancel", comment: "Cancel button text")))
+            } else {
+                switch mode {
+                    case .modal: return AnyView(EmptyView())
+                    case .flow: return AnyView(Text(NSLocalizedString("Back", comment: "Back button text")))
+                }
+            }
+        }
+    }
+    
+    private var editOrDoneButton: some View {
+        let action = {
+            if self.isEditing {
+                // TODO: confirm
+                self.delegate?.save()
+            }
+            self.isEditing.toggle()
+        }
+        return Button( action: action ) {
+            Group {
+                if isEditing {
+                    Text(NSLocalizedString("Done", comment: "Done button text"))
+                } else {
+                    Text(NSLocalizedString("Edit", comment: "Edit button text"))
+                }
+            }
+            .onTapGesture {
+                action()
+            }
+            .onLongPressGesture {
+                self.editButtonDesignVersion = self.editButtonDesignVersion.cycle()
+                AudioServicesPlayAlertSound(kSystemSoundID_Vibrate)
+            }
+        }
+    }
+    
+    private var correctionRangeSection: some View {
+        SectionWithEdit(isEditing: $isEditing,
+                        editButtonDesignVersion: $editButtonDesignVersion,
+                        title: NSLocalizedString("Correction Range", comment: "Correction Range section title"),
+                        footer: EmptyView(),
+                        editAction: { self.delegate?.gotoEdit(therapySetting: TherapySetting.glucoseTargetRange) })
+        {
+            Group {
+                if self.glucoseUnit != nil {
+                    ForEach(self.therapySettings.glucoseTargetRangeSchedule?.items ?? [], id: \.self) { value in
+                        ScheduleRangeItem(time: value.startTime, range: value.value, unit: self.glucoseUnit!, guardrail: Guardrail.correctionRange)
+                    }
+                } else {
+                    DescriptiveText(label: NSLocalizedString("Tap \"Edit\" to add a Correction Range", comment: "Correction Range section edit hint"))
+                }
+            }
+        }
+    }
+    
+    private var temporaryCorrectionRangesSection: some View {
+        SectionWithEdit(isEditing: $isEditing,
+                        editButtonDesignVersion: $editButtonDesignVersion,
+                        title: NSLocalizedString("Temporary Correction Ranges", comment: "Temporary Correction Ranges section title"),
+                        footer: EmptyView(),
+                        editAction: { self.delegate?.gotoEdit(therapySetting: TherapySetting.correctionRangeOverrides) })
+        {
+            ForEach([ CorrectionRangeOverrides.Preset.preMeal, CorrectionRangeOverrides.Preset.workout  ], id: \.self) { preset in
```

Comment:
You can conform CorrectionRangeOverrides.Preset to CaseIterable then use `CorrectionRangeOverrides.Preset.allCases` here